src/Microsoft.AspNet.Razor.Runtime/TagHelpers/RestrictChildrenAttribute.cs

```diff
@@ -0,0 +1,35 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+
+namespace Microsoft.AspNet.Razor.Runtime.TagHelpers
+{
+    /// <summary>
+    /// Restricts children of the <see cref="ITagHelper"/>'s element.
+    /// </summary>
+    [AttributeUsage(AttributeTargets.Class, Inherited = true, AllowMultiple = false)]
+    public class RestrictChildrenAttribute : Attribute
+    {
+        /// <summary>
+        /// Instantiates a new instance of the <see cref="RestrictChildrenAttribute"/> class.
+        /// </summary>
+        /// <param name="tagName">The tag name of the HTML element to allow as children.</param>
+        /// <param name="tagNames">Additional tag names of HTML elements to restrict as children.</param>
```

Comment:
Confusing as this line says `restrict as children` and the previous one says `allow as children`.